plugin/store.go

```diff
@@ -204,6 +232,15 @@ func (ps *Store) Handle(capability string, callback func(string, *plugins.Client
 	}
 }
 
+// RegisterRuntimeOpt stores a list of SpecOpts for the provided capability.
+// These options are applied to the runtime spec before a plugin is started for the specified capability.
```

Comment:
Can we use this method for the customizations we do for volume and graphdriver plugins?

Reply:
We can possibly, but I'd like to do it separately.
The changes I did include here are pretty numerous just because of attempting to achieve my end goal for this PR (do not use a private mount on the plugin dir), all these things had to happen first.